corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js

```diff
@@ -594,10 +594,13 @@ hqDefine("cloudcare/js/form_entry/form_ui", function () {
         let parentForm = getParentForm(self);
         let oneQuestionPerScreen = parentForm.displayOptions.oneQuestionPerScreen !== undefined && parentForm.displayOptions.oneQuestionPerScreen();
 
-        if (!oneQuestionPerScreen && self.isRepetition) {
+        // Header and captions
+        self.needsHeader = oneQuestionPerScreen || self.isRepetition || self.caption() || ko.utils.unwrapObservable(self.caption_markdown);
+        if (self.needsHeader && !oneQuestionPerScreen && self.isRepetition) {
```

Comment:
Just checking, is it correct that `needsHeader` should be `true` if `oneQuestionPerScreen` is `true`? Just because that contradicts with the `!oneQuestionPerScreen` here.

Reply:
Got it, in that case, i don't quite understand. Why are we hiding caption if `needsHeader` is True?

Reply:
Oh i see, the caption is part of the header. I thought they were two separate things.

Maybe `needsHeader` could be renamed to `showHeader` and `!oneQuestionPerScreen && self.isRepetition` can be defined as a `showCaption` variable? Just a thought, not sure if it's any better